src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml/AvoidInfiniteRecursion.xml

```diff
@@ -249,4 +249,86 @@ class IterationsBetter {
      ]]></code>
     </test-code>
 
+    <test-code>
+        <description>Issue484 No violation: false positives on same method name but other class</description>
```

Comment:
Did you also have a test for overloaded methods, as this example (I had added it to the issue a bit later, did you see it?) 

```java
public class Deserializer {

    public Map deserialize(JsonParser p, Context c) {
        ....
        return target;
    }

    public Map deserialize(JsonParser p) {
        Context c = new Context();
        return this.deserialize(p, c);
    }
}
```

Reply:
No, did not see. A new case, overloaded method of the same class

Reply:
I added additional checking whether the number of arguments match. Actually checking if the argument types match is very difficult, I didn't find a reasonable way in XPath. Unfortunately, there doesn't seem to be a way to access Function like in the designer. OTOH calling an overloaded method with the same number of arguments is rather confusing and bad practice IMHO.